stubs/Pygments/pygments/lexer.pyi

```diff
@@ -19,83 +40,137 @@ class LexerMeta(type):
 
 class DelegatingLexer(Lexer):
-    root_lexer: Incomplete
-    language_lexer: Incomplete
+    root_lexer: Lexer
+    language_lexer: Lexer
     needle: Incomplete
-    def __init__(self, _root_lexer, _language_lexer, _needle=..., **options) -> None: ...
+    def __init__(
+        self, _root_lexer: type[Lexer], _language_lexer: type[Lexer], _needle: _TokenType = ..., **options: Any
+    ) -> None: ...
     def get_tokens_unprocessed(self, text: str) -> Iterator[tuple[int, _TokenType, str]]: ...
 
 class include(str): ...
 class _inherit: ...
 
-inherit: Incomplete
+inherit: _inherit
 
-class combined(tuple[Incomplete, ...]):
-    def __new__(cls, *args): ...
-    def __init__(self, *args) -> None: ...
+class combined(tuple[str, ...]):
+    def __new__(cls, *args: str): ...
+    def __init__(self, *args: str) -> None: ...
 
 class _PseudoMatch:
-    def __init__(self, start, text) -> None: ...
-    def start(self, arg=None): ...
-    def end(self, arg=None): ...
-    def group(self, arg=None): ...
-    def groups(self): ...
-    def groupdict(self): ...
+    def __init__(self, start: int, text: str) -> None: ...
+    def start(self, arg=None) -> int: ...
+    def end(self, arg=None) -> int: ...
+    def group(self, arg=None) -> str: ...
+    def groups(self) -> tuple[str]: ...
+    def groupdict(self) -> dict[str, Any]: ...
 
-def bygroups(*args): ...
+def bygroups(
+    *args: _TokenType | Callable[[Lexer, _PseudoMatch, LexerContext], Iterator[tuple[int, _TokenType, str]]]
+) -> Callable[[Lexer, _PseudoMatch, LexerContext], Iterator[tuple[int, _TokenType, str]]]: ...
 
 class _This: ...
 
-this: Incomplete
+this: _This
 
-def using(_other, **kwargs): ...
+def using(
+    _other: _This | Lexer, **kwargs: Any
+) -> Callable[[Lexer, _PseudoMatch, LexerContext], Iterator[tuple[int, _TokenType, str]]]: ...
 
 class default:
-    state: Incomplete
-    def __init__(self, state) -> None: ...
+    state: str
+    def __init__(self, state: str) -> None: ...
 
 class words(Future):
-    words: Incomplete
-    prefix: Incomplete
-    suffix: Incomplete
-    def __init__(self, words, prefix: str = "", suffix: str = "") -> None: ...
-    def get(self): ...
+    words: Sequence[str]
+    prefix: str
+    suffix: str
+    def __init__(self, words: Sequence[str], prefix: str = "", suffix: str = "") -> None: ...
+    def get(self) -> str: ...
 
 class RegexLexerMeta(LexerMeta):
-    def process_tokendef(cls, name, tokendefs=None): ...
-    def get_tokendefs(cls): ...
-    def __call__(cls, *args, **kwds): ...
+    def process_tokendef(
+        cls,
+        name: str,
+        tokendefs: (
+            dict[
+                str,
+                list[
+                    tuple[str, _TokenType | Iterator[tuple[int, _TokenType, str]]]
+                    | tuple[str, _TokenType | Iterator[tuple[int, _TokenType, str]], str]
+                ],
+            ]
+            | None
+        ) = None,
```

Comment:
I haven't looked at this in detail, but this could be problematic. Due to the invariance of `list`, this will reject calls like this:

```python
x.process_tokendef("", {"foo": [("", "")]})
```

In this case, the `dict` could be inferred to have type `dict[str, list[tuple[str, str]]]`, but `list[tuple[str, str]]` is incompatible with `list[tuple[str, str, str]]`.

Same below.